### PR TITLE
fix the docs and simplify the implementation of unsigned wrapping ops

### PR DESCRIPTION
**Pull request by oli-obk**
r? @steveklabnik

### CODE REVIEW

**Comment by steveklabnik** [comment]
The Travis failure looks legit here.

**Comment by oli-obk** [comment]
whoops. `check-doc` now passes locally.

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -1278,15 +1275,9 @@ macro_rules! uint_impl {
...
-        ///
-        /// The only case where such wrapping can occur is when one
-        /// divides `MIN / -1` on a signed type (where `MIN` is the
-        /// negative minimal value for the type); this is equivalent
-        /// to `-MIN`, a positive value that is too large to represent
-        /// in the type. In such a case, this function returns `MIN`
-        /// itself.
+        /// Wrapping (modular) division. Computes `self / other`.
+        /// This operation only makes sense on signed types.
+        /// It is here for completeness only.
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: This comment makes no sense to me, why is it 'complete' to include something that doesn't make sense?

Comment (oli-obk) [inline src/libcore/num/mod.rs]: Yea, I should probably word that differently. Wrapped division on unsigned types is just normal division. There's no way wrapping could ever happen. So this function is here, so all operations are accounted for in the wrapping operations.

Comment (steveklabnik) [inline src/libcore/num/mod.rs]: Yeah, that sounds very reasonable, gotta find a better way to say that :0

**Comment by alexcrichton** [comment]
We may want to coordinate with https://github.com/rust-lang/rust/pull/30466, looks like these are all touching similar surface area?

(I'll try to get around to rebasing and approving that today or tomorrow)

**Comment by oli-obk** [comment]
I'll just rebase after that PR, looks like just minor overlaps

**Comment by alexcrichton** [comment]
Sure, sounds good to me

**Comment by oli-obk** [comment]
apparently there was no overlap

**Comment by oli-obk** [comment]
I changed the wording, this is ready now.

**Comment by steveklabnik** [comment]
So, I am confused here with how said coordination went or should go, so r? @alexcrichton

**Comment by alexcrichton** [comment]
@bors: r+ 84a5790f3a60815d42f81aa6adca6fc015524613

Thanks!

**Pull request merged**